script.kodi.windows.update/service.py

```diff
@@ -38,38 +39,52 @@ def log(msg, level=xbmc.LOGDEBUG):
 
 class Service(object):
     def __init__(self):
-        try:
-            self.getBuild()
-            self.getVersion()
-            self.lastPath = REAL_SETTINGS.getSetting("LastPath") # CACHE = Keep last download, CLEAN = Remove all downloads
-            if not CACHE and CLEAN and xbmcvfs.exists(self.lastPath):
-                log('Service, removing last download')
-                xbmcvfs.delete(self.lastPath)
-                xbmcgui.Dialog().notification(ADDON_NAME, LANGUAGE(30007), ICON, 4000)
-        except Exception as e: log("Service Failed! " + str(e), xbmc.LOGERROR)
+        self.setSettings()
+        lastPath = REAL_SETTINGS.getSetting("LastPath") # CACHE = Keep last download, CLEAN = Remove all downloads
+        if not CACHE and CLEAN and xbmcvfs.exists(lastPath): self.deleteLast(lastPath)
+        
                 
+    def deleteLast(self, lastPath):
+        log('deleteLast')
+        #some file systems don't release the file lock instantly.
+        for count in range(3):
+            if xbmc.Monitor().waitForAbort(1): return 
+            try: 
+                if xbmcvfs.delete(lastPath): return
+            except: pass
+                
+
+    def setSettings(self):
+        log('setSettings')
+        [func() for func in [self.getBuild,self.getPlatform,self.getVersion]]
+              
+              
+    def getBuild(self):
+        log('getBuild')
+        REAL_SETTINGS.setSetting("Build",json.dumps(json.loads(xbmc.executeJSONRPC(VER_QUERY) or '').get('result',{}).get('version',{})))
         
-    def getBuild(self): 
+        
+    def getPlatform(self): 
+        log('getPlatform')
         count = 0
-        build = ''
         try:
-            while not xbmc.Monitor().abortRequested() and count < 3:
-                count += 1
+            while not xbmc.Monitor().abortRequested() and count < 15:
```

Comment:
reuse monitor?

Reply:
It's a lazy method, probably could have written better; with a single Monitor call.